ebos/ecloutputblackoilmodule.hh

```diff
@@ -1038,6 +1099,324 @@ public:
+				const auto& defunctWellNames = simulator_.vanguard().defunctWellNames(); 
+				if (defunctWellNames.find(wname) != defunctWellNames.end()) {
+					continue;
+				}
+
+                const auto& well = schedule.getWell2(wname, reportStepNum);			
+					
+				// Ignore injector wells  
+				if (well.isInjector()){ 
+					continue;
+				}
+	
+				tmp_names[0] = wname;//WellProdDataType::WellName
+	
+				
+				auto wName = static_cast<std::string>(wname);                
+                auto get = [&st, &wName](const std::string& vector)
+                {
+                const auto key = vector + ':' + wName;
+
+                return st.has(key) ? st.get(key) : 0.0;
+                };
+				
+				const auto& controls = well.productionControls(st);
+				using CMode = ::Opm::Well2::ProducerCMode;
+				
+				auto fctl = [](const auto wmctl) -> std::string
+				{
+					switch (wmctl) {
+				    case CMode::ORAT: return "ORAT";
+					case CMode::WRAT: return "WRAT";
+					case CMode::GRAT: return "GRAT";
+					case CMode::LRAT: return "LRAT";
+					case CMode::RESV: return "RESV";
+					case CMode::THP:  return "THP";
+					case CMode::BHP:  return "BHP";
+					case CMode::CRAT: return "CRate";
+					case CMode::GRUP: return "GRUP";
+					default:
+					{
+						return "none";
+					}
+					}
+			    };
+
+				tmp_names[1] = fctl(controls.cmode); //WellProdDataType::CTRLMode
+	
+				tmp_values[0] = well.getHeadI() + 1;//WellProdDataType::WellLocationi
+				tmp_values[1] = well.getHeadJ() + 1;//WellProdDataType::WellLocationj
+				tmp_values[2] = get("WOPR"); //WellProdDataType::OilRate 
+				tmp_values[3] = get("WWPR"); //WellProdDataType::WaterRate
+				tmp_values[4] = get("WGPR"); //WellProdDataType::GasRate
+				tmp_values[5] = get("WVPR"); //WellProdDataType::FluidResVol
+				tmp_values[6] = get("WWCT"); //WellProdDataType::WaterCut
+				tmp_values[7] = get("WGOR"); //WellProdDataType::GasOilRatio
+				tmp_values[8] = get("WWPR")/get("WGPR"); //WellProdDataType::WaterGasRatio					
+				tmp_values[9] = get("WBHP"); //WellProdDataType::BHP
+				tmp_values[10] = get("WTHP"); //WellProdDataType::THP
+				//tmp_values[11] = 0; //WellProdDataType::SteadyStatePI //			
+				
+				outputProductionReport_(tmp_values, tmp_names, forceDisableProdOutput); 	
+				
+			}
+		}
+    }
+
+    
+    // write injection report to output 
+    void outputInjLog(size_t reportStepNum, const bool substep, bool forceDisableInjOutput)
+    {
+		if (!substep) {		
+			ScalarBuffer  tmp_values(WellInjDataType::numWIValues, 0.0);
+			StringBuffer  tmp_names(WellInjDataType::numWINames, "");
+			outputInjectionReport_(tmp_values, tmp_names, forceDisableInjOutput);
+			
+			const auto& st = simulator_.vanguard().summaryState();
+			const auto& schedule = simulator_.vanguard().schedule();
+            for (const auto& group_name: schedule.groupNames()) {
+                std::cout << group_name; //well.groupName()
+            }
```

Comment:
Is this (i.e., printing the group names to `cout`) something that's left over from earlier debugging?  If so, I think it shuold be removed before we merge this into master.

Reply:
Yes, removed.